src/components/Tooltip/Tooltip.js

```diff
@@ -0,0 +1,94 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import Portal from '../Portal/Portal';
+import styles from './Tooltip.scss';
+
+class Tooltip extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      visible: false,
+    };
+
+    this.showTooltip = this.showTooltip.bind(this);
+    this.hideTooltip = this.hideTooltip.bind(this);
+  }
+
+  showTooltip() {
+    const SPACE = 8;
+    const WIDTH = 256;
+
+    const style = {
+      width: WIDTH,
+      borderColor: this.props.borderColor,
+    };
+    const dimensions = this.el.getBoundingClientRect();
+
+    // center align the tooltip by taking both the target and tooltip widths into account
+    style.left = (dimensions.left + (dimensions.width / 2)) - (WIDTH / 2);
+    // make sure it doesn't stick off the left side of the page
+    style.left = Math.max(SPACE, style.left);
+    // or off the right ...
+    style.left = Math.min(style.left, document.body.clientWidth - WIDTH - SPACE);
+
+    if (dimensions.top < window.innerHeight / 2) { // the top half of the page
+      // on the top half of the page, position the top of the tooltip
+      // just below the target (it will stretch downwards)
+      style.top = dimensions.top + dimensions.height + SPACE;
+    } else {
+      // on the bottom half, set the bottom of the tooltip
+      // just above the top of the target (it will stretch upwards)
+      style.bottom = (window.innerHeight - dimensions.top) + SPACE;
+    }
+
+    this.setState({
+      visible: true,
+      style,
+    });
+  }
+
+  hideTooltip() {
+    this.setState({ visible: false });
+  }
+
+  render() {
+    const { props, state } = this;
+
+    return (
+      <span // a span so it's valid HTML no matter where it's used
+        onMouseOver={this.showTooltip}
+        onMouseOut={this.hideTooltip}
```

Comment:
This works great on desktop but on mobile you can end up with a floating, out-of-context tooltip:

![image](https://user-images.githubusercontent.com/165087/32420625-331c399e-c2e1-11e7-9ea4-5643f0679dd5.png)

It can be dismissed with a touch, but would be nice to just hide it on scroll, like this: 

```
  componentDidMount() {
    window.addEventListener('scroll', this.hideTooltip);
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.hideTooltip);
  }
```

Reply:
Holy crap why didn't I think of that?! I messed around for ages listening for touchstart to try and get it to hide and it was being a pain.

Reply:
Done, FYI I add/remove it only when showing/hiding a tooltip (otherwise you get 20 scroll listeners when all the tooltips mount).